cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs

```diff
@@ -715,6 +723,18 @@ exAvSrcPkg ex =
     isMissingUpperBound pc = case C.explanation pc of
       C.MissingUpperBounds{} -> True
       _ -> False
+    isTrailingZeroUpperBound :: C.PackageCheck -> Bool
+    isTrailingZeroUpperBound pc = case C.explanation pc of
+      C.TrailingZeroUpperBounds{} -> True
+      _ -> False
+    isLeqUpperBound :: C.PackageCheck -> Bool
+    isLeqUpperBound pc = case C.explanation pc of
+      C.LEQUpperBounds{} -> True
+      _ -> False
+    isgtLowerBound :: C.PackageCheck -> Bool
```

Comment:
Is there a reason to hav multiple functions instead of one? (`isBoundCheck` or similar)

Reply:
I didn't think too deeply about this and went with the flow, following the pattern already there.